packages/styled-base/src/index.js

```diff
@@ -43,6 +43,20 @@ let createStyled: CreateStyled = (tag: any, options?: StyledOptions) => {
   const isReal = tag.__emotion_real === tag
   const baseTag = (isReal && tag.__emotion_base) || tag
 
+  const displayName =
+    identifierName !== undefined
+      ? identifierName
+      : `Styled(${
+          typeof baseTag === 'string'
+            ? baseTag
+            : baseTag.displayName || baseTag.name || 'Component'
+        })`
+
+  const injectDisplayName = Component => {
+    Component.displayName = Component.displayName || displayName
```

Comment:
there is no way for the `Component` to have a `.displayName` as it's always the anonymous function